docs/source/learn/trace-vs-execution-time.md

```diff
@@ -0,0 +1,176 @@
+# Tracing Time vs. Execution Time in PyTorch/XLA
+
+When working with PyTorch/XLA, it's essential to understand that operations on
+XLA tensors are not typically executed immediately in the way they are with
+standard PyTorch tensors on CPU or CUDA devices (which operate in "eager mode").
+PyTorch/XLA employs a "lazy execution" model. This means that when you write
+PyTorch code using XLA tensors, you are primarily defining or tracing a
+computation graph. The actual execution of this graph on the XLA device (like a
+TPU) is deferred until a specific trigger point.
+
+This leads to two distinct types of "time" to consider:
+
+1. Tracing Time: The period during which PyTorch/XLA records your operations and
+builds this graph on the host (CPU).
+
+1. Device Time: The period during which the XLA device (e.g., TPU) actually
+performs the computations. This can be further broken down:
+
+   - Compilation Time: The time XLA takes to compile the traced graph into
+   optimized device code. This is most significant on the first execution of a
+   new graph or if the graph changes.
+
+   - Execution Time: The time the XLA device spends running the compiled code.
+
+**If you measure the duration of code that only performs XLA operations without an
+explicit instruction to wait for the device, you are primarily measuring this
+tracing time plus Python overhead. This is NOT an accurate measure of your
+model's performance on the XLA device.**
+
+## Tracing: Defining the Computation (Host-Side Activity)
+
+When you write PyTorch code using XLA tensors (e.g., tensors on a TPU),
```

Comment:
This paragraph is essentially the same as the one directly above. I think we can remove the paragraph above.